frontend/packages/dev-console/integration-tests/features/topology/topology-smoke_test.feature

```diff
@@ -0,0 +1,165 @@
+Feature: Topology smoke test
+	As a user, I want to verify topology feature    
+
+@regression, @smoke
+Scenario: Empty state of topology
+   Given user is in administrative perspective
+   When user goes to developer perspective
+   And user creates a new project
+   Then user sees Add page with message on the top "No workloads found"
+
+@regression, @smoke
+Scenario: Topology with workloads
+   Given user is in administrative perspective
+   When user goes to developer perspective
+   And user selects a existing project from project list with existing workloads
+   Then user sees different workloads in topology chart area
+
+@regression, @smoke
+Scenario: Visual for deployment/deployment-config
+   Given user is in topology with deployment workload
+   When user checks nodes and the decorators associated with them
+   And nodes are circular shaped with builder image in them
+   And pod ring associated with node are present around node with color according to the pod status
+   And deployment can have application url on top-right of the node 
+   And user sees edit source code decorator is on bottom right of the node which can lead to github or che workspace
+   And user sees build decorator on bottom left which will take user to either build tab or pipeline depending on pipeline associated with them 
+   Then user checks node label having "D" for deployment and then name of node
+
+@regression, @smoke
+Scenario: Visual for deployment-config
+   Given user is in topology with deployment-config workload
+   When user checks nodes and the decorators associated with them
+   And nodes are circular shaped with builder image in them
+   And pod ring associated with node are present around node with color according to the pod status
+   And deployment-config can have application url on top-right of the node 
+   And user sees edit source code decorator is on bottom right of the node which can lead to github or che workspace
+   And user sees build decorator on bottom left which will take user to either build tab or pipeline depending on pipeline associated with them 
+   Then user checks node label having "DC" for deployment-config and then name of node
+
+@regression, @smoke
+Scenario: Visual for knative service with no revision
+   Given user is in topology with knative workload
+   When user checks nodes and the decorators associated with them
+   And user can view knative service are rectangular shaped with round corners
```

Comment:
If not performing any action, change it to then statement.

Reply:
Same as above